src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs

```diff
@@ -22,40 +25,59 @@ public ImageSourceServiceProvider(IImageSourceServiceCollection collection, ISer
-					return genericConcreteType;
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		Type CreateImageSourceServiceType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type type)
+		{
+			var genericConcreteType = MakeGenericType(ImageSourceServiceType, type);
 
-				return ImageSourceServiceType.MakeGenericType(GetImageSourceType(type));
-			});
+			if (genericConcreteType != null && GetServiceDescriptor(genericConcreteType) != null)
+				return genericConcreteType;
 
-		public Type GetImageSourceType(Type imageSource) =>
+			return MakeGenericType(ImageSourceServiceType, GetImageSourceType(type));
+		}
+
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = TrimmerHelper.ConcurrentDictionary)]
+		[UnconditionalSuppressMessage("Trimming", "IL2111", Justification = TrimmerHelper.MakeGenericType)]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		public Type GetImageSourceType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type imageSource) =>
 			_imageSourceCache.GetOrAdd(imageSource, CreateImageSourceTypeCacheEntry);
 
-		Type CreateImageSourceTypeCacheEntry(Type type)
+		[UnconditionalSuppressMessage("Trimming", "IL2065", Justification = "'directInterface.GetInterface(fullName)' can not be statically determined and may not meet 'DynamicallyAccessedMembersAttribute' requirements.")]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		Type CreateImageSourceTypeCacheEntry([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type type)
 		{
-			if (type.IsInterface)
+			var fullName = ImageSourceInterfaceType.FullName;
+			if (!string.IsNullOrEmpty(fullName))
 			{
-				if (type.GetInterface(ImageSourceInterface) != null)
-					return type;
-			}
-			else
-			{
-				foreach (var directInterface in type.GetInterfaces())
+				if (type.IsInterface)
+				{
+					if (type.GetInterface(fullName) != null)
+						return type;
+				}
+				else
 				{
-					if (directInterface.GetInterface(ImageSourceInterface) != null)
-						return directInterface;
+					foreach (var directInterface in type.GetInterfaces())
+					{
+						if (directInterface.GetInterface(fullName) != null)
+							return directInterface;
+					}
 				}
 			}
 
 			throw new InvalidOperationException($"Unable to find the image source type because none of the interfaces on {type.Name} were derived from {nameof(IImageSource)}.");
 		}
+
+		[UnconditionalSuppressMessage("Trimming", "IL2055", Justification = TrimmerHelper.MakeGenericType)]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+		static Type MakeGenericType ([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type a, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type b) => a.MakeGenericType(b);
```

Comment:
Why is `a` marked with `Interfaces`?
I must admit I don't understand the image source services, but the "service" doesn't seem to need all interfaces, only the image source needs these.

There might also be a simpler version of all this. If the trimmer sees the type as instantiated and it also sees the interface in question used, then it will keep the interface on the type - regardless if it saw a direct access to that interface from that type.
So the key here is the image source type is marked as instantiated (its constructors are preserved) I think.

Reply:
I think I was getting the warning from this method:

https://github.com/dotnet/maui/blob/58d0b1481703850ee71ac01f03c336cef44bf0a4/src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs#L54-L77

Let me try taking out `Interfaces`, and see, thanks.